### ISSUE TITLE
RecursionError for some code

### ISSUE THREAD

**Issue by albertz** [PyCQA/pylint#2836]
I'm not sure if this is a duplicate of #1452 or not, but I thought I post it here as a new issue, as #1452 is closed.

Example code:
```
class Data:
    def __init__(self):
        self.shape = [None, 17]

    @property
    def ndim(self):
        return len(self.shape)

    def copy_move_axis(self, old_axis):
        if old_axis < 0:
            old_axis += self.ndim
            assert old_axis >= 0
        assert 0 <= old_axis < self.ndim

        new_shape = [None] * self.ndim
        self.shape = new_shape
        return self
```

This results in an infinite recursion (very much as in #1452), with Pylint 2.3.1, Astroid 2.2.5, Python 3.6.
Maybe the bug actually is in Astroid, so not sure if I should report here or for Astroid.

**Comment by albertz** [comment]
Maybe #2736 is also related.

**Comment by PCManticore** [comment]
Thank you @albertz I can reproduce the bug

**Comment by PCManticore** [comment]
Doesn't seem to reproduce any longer using the `master` branch of both `astroid` and `pylint`.
